File: pdftract-dotnet/src/Pdftract/Models/Fingerprint.cs
using MessagePack;

namespace Pdftract.Models;

/// <summary>
/// Represents a document fingerprint for identification.
/// </summary>
[MessagePackObject]
public record Fingerprint
{
    /// <summary>
    /// Document hash value.
    /// </summary>
    [Key(0)]
    public required string Hash { get; init; }

    /// <summary>
    /// Document size in bytes.
    /// </summary>
    [Key(1)]
    public required long Size { get; init; }

    /// <summary>
    /// Number of pages in the document.
    /// </summary>
    [Key(2)]
    public required int PageCount { get; init; }
}

File: sdk/php/src/Pdftract/Models/Fingerprint.php
<?php

declare(strict_types=1);

namespace Jedarden\Pdftract\Models;

/**
 * Readonly fingerprint model
 *
 * Simple readonly representation of a PDF document fingerprint
 */
class Fingerprint
{
    /**
     * @param string $id Unique fingerprint identifier
     * @param int $pageCount Total number of pages in the document
     * @param string $contentHash Hash of the document content
     * @param string $structureHash Hash of the document structure
     */
    public function __construct(
        public readonly string $id,
        public readonly int $pageCount,
        public readonly string $contentHash,
        public readonly string $structureHash
    ) {}
}
